dcos/http.py

```diff
@@ -180,3 +220,29 @@ def delete(url, **kwargs):
 def silence_requests_warnings():
     """Silence warnings from requests.packages.urllib3.  See DCOS-1007."""
     requests.packages.urllib3.disable_warnings()
+
+
+def _get_http_auth_credentials(response):
+    """Get username/password for HTTP Basic Auth
+
+    :param response: requests.response
+    :type response: requests.Response
+    :returns: process return code
+    :rtype: int
+    """
+
+    parsed_url = urlparse(response.url)
+    hostname = parsed_url.hostname
+    if response.headers.get('www-authenticate') == 'Basic realm="Restricted"':
+        username = parsed_url.username
+        if username is None:
+            sys.stdout.write('Username: ')
+            sys.stdout.flush()
+            username = sys.stdin.readline().strip().lower()
+        password = getpass.getpass()
+        AUTH_HOSTS[hostname] = HTTPBasicAuth(username, password)
+    else:
+        raise DCOSException("We don't support this authentication. {}".format(
+            response.content))
+
+    return 0
```

Comment:
I think this should return a `BasicAuth` this is going to be useful when we replace this with a long running process that securely stores and returns user credentials. Thoughts?